### ISSUE TITLE
[question] Unrelated changes in conandata.yml affect RREV

### ISSUE THREAD

**Issue by db4** [conan-io/conan#10526]
- [x] I've read the [CONTRIBUTING guide](https://github.com/conan-io/conan/blob/develop/.github/CONTRIBUTING.md).

When I add a new version to `conandata.yml` it changes RREVs of all over versions listed there (because RREV depends on the hash of the whole file) and so all existing binary packages become obsolete. This is very inconvenient. Maybe Conan could use only the current version subtree in `conandata.yml` when calculating RREV?

**Comment by memsharded** [comment]
Hi @db4 

This is expected. The current solution is using a hook (check the one we are using in conan-center: https://github.com/conan-io/hooks/blob/a52cef5ffed0ff11bed757c42e66a95f792572b8/hooks/conan-center.py#L762) that implement exactly that.

**Comment by db4** [comment]
Hi @memsharded 

> This is expected. The current solution is using a hook (check the one we are using in conan-center: https://github.com/conan-io/hooks/blob/a52cef5ffed0ff11bed757c42e66a95f792572b8/hooks/conan-center.py#L762) that implement exactly that.

Didn't know that, thanks. Maybe this should be the core functionality in Conan 2.0?

**Comment by memsharded** [comment]
> Didn't know that, thanks. Maybe this should be the core functionality in Conan 2.0?

Yes, totally, this is why I assigned it as "look into" for next 2.0.0-alpha4, that basically means "having a look and consider something else from Conan side"
